wp-content/themes/core/assets/js/admin/block-animation.js

```diff
@@ -0,0 +1,413 @@
+			: `${ props.className } tribe-animation-style-${ animationStyle } is-animated-on-scroll`;
+
+	if ( animationSpeed !== undefined && animationSpeed ) {
+		props.style = {
+			...props.style,
+			'--tribe-animation-speed': animationSpeed,
+		};
+	}
+
+	if ( animationDelay !== undefined && animationDelay ) {
+		props.style = {
+			...props.style,
+			'--tribe-animation-delay': animationDelay,
+		};
+	}
+
+	if ( animationEasing !== undefined && animationEasing ) {
+		props.style = {
+			...props.style,
+			'--tribe-animation-easing': animationEasing,
+		};
+	}
+
+	if ( animationTrigger !== undefined && animationTrigger ) {
+		props.className = `${ props.className } tribe-animate-multiple`;
+	}
+
+	return props;
+};
+
+/**
+ * @function animationControls
+ *
+ * @description creates component that overrides the edit functionality of the block with new animation controls
+ */
+const animationControls = createHigherOrderComponent( ( BlockEdit ) => {
+	return ( props ) => {
+		const { attributes, setAttributes, isSelected, name } = props;
+
+		// return default Edit function if block isn't in the includes array
+		if ( state.includes.length > 0 && ! state.includes.includes( name ) ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		// return default Edit function if block is in the excludes array
+		if ( state.excludes.length > 0 && state.excludes.includes( name ) ) {
+			return <BlockEdit { ...props } />;
+		}
+
+		const {
+			animationStyle,
+			animationSpeed,
+			animationDelay,
+			animationEasing,
+			animationTrigger,
+			animationPosition,
+		} = attributes;
+
+		let blockClass =
+			attributes.className !== undefined ? attributes.className : '';
+		const blockStyles = { ...props.style };
+
+		if ( animationStyle !== undefined && animationStyle !== 'none' ) {
+			blockClass =
+				animationPosition !== undefined && animationPosition
+					? `${ blockClass } tribe-animation-style-${ animationStyle } is-animated-on-scroll-full`
+					: `${ blockClass } tribe-animation-style-${ animationStyle } is-animated-on-scroll`;
+
+			if ( animationSpeed !== undefined && animationSpeed ) {
+				blockStyles[ '--tribe-animation-speed' ] = animationSpeed;
+			}
+
+			if ( animationDelay !== undefined && animationDelay ) {
+				blockStyles[ '--tribe-animation-delay' ] = animationDelay;
+			}
+
+			if ( animationEasing !== undefined && animationEasing ) {
+				blockStyles[ '--tribe-animation-easing' ] = animationEasing;
+			}
+
+			if ( animationTrigger !== undefined && animationTrigger ) {
+				blockClass = `${ blockClass } tribe-animate-multiple`;
+			}
+		}
+
+		const blockProps = {
+			...props,
+			attributes: {
+				...attributes,
+				className: blockClass,
+			},
+			style: blockStyles,
+		};
+
+		return (
+			<Fragment>
+				<div style={ blockStyles } className={ blockClass }>
+					<BlockEdit { ...blockProps } />
+					{ isSelected && (
+						<InspectorControls>
+							<PanelBody
```

Comment:
@GeoffDusome it might be nice to start our own `/components` directory and pull this whole piece out into its own component. That way if we are creating custom blocks in the future we can just do something like below and not have to add/edit this everywhere.

```
<InspectorControls>
    <AnimationControls />
</ InspectorControls>